app/controllers/sessions_controller.rb

```diff
@@ -19,75 +18,11 @@ def callback
       state: params.require(:state),
     ).to_h
 
-    cookie_consent =
-      case params[:cookie_consent]
-      when "accept"
-        update_saved_cookie_consent = !callback["cookie_consent"]
-        "accept"
-      when "reject"
-        update_saved_cookie_consent = callback["cookie_consent"]
-        "reject"
-      else
-        update_saved_cookie_consent = false
-        callback["cookie_consent"] ? "accept" : "reject"
-      end
-
-    if callback.key?("cookie_consent") && callback.key?("feedback_consent") && (callback["cookie_consent"].nil? || callback["feedback_consent"].nil?)
-      # slightly hacky way of passing the session header along in the
```

Comment:
I guess we don't need this any more becuase now we can just "properly log the user in"?

Reply:
Yes exactly - the user always gets logged in now when they hit this endpoint (assuming they have a valid code etc.) so there's no need to mess with the session header or check the previously stored attributes at all. 

Now all this endpoint does is check their code, fetch `userinfo`, create an account if they're new and redirect them onto the page they came from. So much simpler!